src/thunderbots/software/ai/hl/stp/tactic/tactic.cpp

```diff
@@ -0,0 +1,38 @@
+#include "ai/hl/stp/tactic/tactic.h"
+
+Tactic::Tactic(const Robot &robot)
+    : robot(robot),
+      intent_sequence(boost::bind(&Tactic::calculateNextIntentWrapper, this, _1))
+{
+}
+
+bool Tactic::done() const
+{
+    return !static_cast<bool>(intent_sequence);
+}
+
+std::unique_ptr<Intent> Tactic::getNextIntent()
+{
+    // If the coroutine "iterator" is done, the calculateNextIntent function has completed
+    // and therefore the Tactic is done, so we return a null pointer
```

Comment:
``` suggestion
  // and therefore the Tactic is done, so we return an empty optional
```

Not super sure if `empty` is the right word here, but it's not really a null pointer